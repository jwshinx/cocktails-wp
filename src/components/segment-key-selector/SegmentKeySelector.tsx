import React from 'react'

import SegmentKeySelectorOptions from './SegmentKeySelectorOptions'
import {
  SegmentKeyType,
  AlcoholicSegmentKey,
  CategorySegmentKey,
  IngredientSegmentKey,
  GlassSegmentKey,
  CocktailNameSegmentKey,
  FirstLetterSegmentKey,
  OrdinaryDrinkValue,
  HighballGlassValue,
} from '../../models/cocktail'

interface SegmentKeySelectorProps {
  onSegmentKeyChange: (keyType: SegmentKeyType) => void
  onSegmentValueChange: (value: string) => void
  segmentType: string
  segmentValue: string
}

const SearchByKeyPairs = [FirstLetterSegmentKey, CocktailNameSegmentKey]

const FilterByKeyPairs = [
  AlcoholicSegmentKey,
  CategorySegmentKey,
  IngredientSegmentKey,
  GlassSegmentKey,
]

const SegmentKeySelector = (props: SegmentKeySelectorProps): JSX.Element => {
  const { onSegmentValueChange, onSegmentKeyChange, segmentType } = props

  const onSelectChange = (value: string): void => {
    if (value === 'cocktailName') {
      onSegmentKeyChange(CocktailNameSegmentKey)
    } else if (value === 'glass') {
      onSegmentKeyChange(GlassSegmentKey)
      onSegmentValueChange(HighballGlassValue.value)
    } else if (value === 'category') {
      onSegmentKeyChange(CategorySegmentKey)
      onSegmentValueChange(OrdinaryDrinkValue.value)
    } else if (value === 'firstLetter') {
      onSegmentKeyChange(FirstLetterSegmentKey)
      onSegmentValueChange('a')
    } else if (value === 'ingredient') {
      onSegmentKeyChange(IngredientSegmentKey)
    } else if (value === 'alcoholic') {
      onSegmentKeyChange(AlcoholicSegmentKey)
      onSegmentValueChange('alcoholic')
    }
  }

  return (
    <div className="form-group">
      <label htmlFor="segment-key">Enter category of search</label>
      <select
        className="form-select mb-3"
        id="segment-key"
        // onChange={(event) => onSelectChange(event.target.value as keyof T)}
        onChange={(event) => {
          onSelectChange(event.target.value)
        }}
      >
        {segmentType === 'filter' ? (
          <SegmentKeySelectorOptions items={FilterByKeyPairs} />
        ) : (
          <SegmentKeySelectorOptions items={SearchByKeyPairs} />
        )}
      </select>
    </div>
  )
}

export default SegmentKeySelector
