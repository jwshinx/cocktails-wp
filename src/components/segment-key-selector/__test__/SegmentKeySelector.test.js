/**
 * @jest-environment jsdom
 */

import React from 'react'
// import ReactDOM from 'react-dom'
// import { render } from '@testing-library/react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

import SegmentKeySelector from '../SegmentKeySelector'

it('should render label "Enter category of search"', () => {
  render(<SegmentKeySelector />)
  const elem = screen.getByText(/Enter category of search/)
  expect(elem).toBeInTheDocument()
})

it('should render combobox', () => {
  render(<SegmentKeySelector />)
  const elem = screen.getByRole('combobox')
  expect(elem).toBeInTheDocument()
})

it('should snapshot SegmentKeySelectorOptions', () => {
  const tree = renderer.create(<SegmentKeySelector />).toJSON()
  expect(tree).toMatchSnapshot()
})
