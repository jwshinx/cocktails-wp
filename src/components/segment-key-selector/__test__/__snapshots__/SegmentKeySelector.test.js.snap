// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot SegmentKeySelectorOptions 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="segment-key"
  >
    Enter category of search
  </label>
  <select
    className="form-select mb-3"
    id="segment-key"
    onChange={[Function]}
  >
    <option
      value="firstLetter"
    >
      First Letter
    </option>
    <option
      value="cocktailName"
    >
      Cocktail Name
    </option>
  </select>
</div>
`;
